module/solutions/helper.js

```diff
@@ -2434,52 +2490,82 @@ module.exports = class SolutionsHelper {
 	 * @param {Number} pageNo - Page No.
 	 * @param {String} search - Search text.
 	 * @param {String} filter - filter text.
-	 * @param {String} entityId - entity id.
+	 * @param {Object} bodyData - request body data
 	 * @returns {Object}
 	 */
 
-	static assignedProjects(userId, search, filter, pageNo, pageSize, entityId = '') {
+	static assignedProjects(userId, search, filter, pageNo, pageSize, bodyData) {
 		return new Promise(async (resolve, reject) => {
 			try {
 				let query = {
 					isDeleted: false,
 				}
-
-				if (entityId !== '') {
-					query['entityId'] = entityId // to fetch projects based on entityId key
-				} else {
-					query['userId'] = userId // to fetch projects based on userId key
-				}
-
 				let searchQuery = []
 
 				if (search !== '') {
 					searchQuery = [{ title: new RegExp(search, 'i') }, { description: new RegExp(search, 'i') }]
 				}
-
-				if (filter && filter !== '') {
-					if (filter === CONSTANTS.common.CREATED_BY_ME) {
-						query['referenceFrom'] = {
-							$ne: CONSTANTS.common.LINK,
-						}
-						query['isAPrivateProgram'] = {
-							$ne: false,
-						}
-					} else if (filter == CONSTANTS.common.ASSIGN_TO_ME) {
-						query['isAPrivateProgram'] = false
-					} else {
-						query['$or'] = [
+				let requestedData = _.cloneDeep(bodyData)
+				if (process.env.SUBMISSION_LEVEL == 'ENTITY' && requestedData.hasOwnProperty('entityId')) {
+					// use queryBasedOnRoleAndLocation function to form query for acl.visibility = SCOPE projects
+					let queryData = await this.queryBasedOnRoleAndLocation(
+						_.omit(requestedData, ['entityId']),
+						'',
+						'acl'
+					)
+					delete queryData.data.isReusable
```

Comment:
@Prajwal17Tunerlabs why we are deleting all these filters,  in every scenario these are common key for filtering right?

Reply:
yes, @VISHNUDAS-tunerlabs  got it, i have reverted the code